services/spar/src/Spar/Run.hs

```diff
@@ -74,7 +76,26 @@ mkLogger opts = Log.new $ Log.defSettings
   & Log.setLogLevel (toLevel $ saml opts ^. SAML.cfgLogLevel)
   & Log.setOutput Log.StdOut
   & Log.setFormat Nothing
-  & Log.setNetStrings (logNetStrings opts)
+  & Log.setRenderer (removeNewline rndrRaw)
+  where
+    rndrRaw :: Log.Renderer
+    rndrRaw = if logNetStrings opts
+      then Log.renderNetstr
+      else Log.renderDefault
+
+-- | Many logging processors handle newlines poorly.  Instead of showing discipline and not
+-- injecting any newlines into your log messages and making sure nobody else does, you can
+-- choose to call `noNewlines` on your renderer that replaces all consecutive whitespace in
+-- the output by a single ' '.
```

Comment:
either the docs or the implementation are lying.  what we want to happen is this:

```
"\t  \n" -> " "
```

perhaps call it `canonicalizeWhitespace` and implement it as `LBS.unwords . LBS.words`?